### PR TITLE
style: start lint/format for src files

### PR DESCRIPTION
**Pull request by kjin**
This change applies `gts` rules for linting, formatting, and tsc compiler options for source files (not including plugins).

Note on the several tsconfig files:

We want files matching `*.ts` and `src/*.ts` to match the TS style guide, but files that are not fully converted (plugins and tests) should be excluded. To accomplish this I have a `tsconfig.full.json` which compiles _all_ files with non-strict rules, and `tsconfig.json` to additionally compile source files under strict rules. Eventually, `tsconfig.full.json` will disappear.

This works well with `gts check` because `gts` will read only `tsconfig.json` to get a list of files to lint and format.

### CODE REVIEW

File: scripts/index.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ async function run(steps: string[]) {
         'npm',
         [
           'run',
-          ...moduleAndArgs.slice(1)
+          moduleAndArgs.slice(1).join('-')
```
Comment (ofrobots) [inline scripts/index.ts]: This seems like a logical rather than style change. Intentional?

File: src/trace-api.ts [hunk 2]
```diff
@@ -75,36 +76,37 @@ export class TraceAgent implements TraceAgentInterface {
   public readonly constants = Constants;
   public readonly labels = TraceLabels;
 
-  private pluginName_: string;
-  private logger_: Logger;
-  private config_: TraceAgentConfig;
+  private pluginName: string;
```
Comment (ofrobots) [inline src/trace-api.ts]: This is not semver major?

File: tsconfig.json [hunk 3]
```diff
@@ -1,18 +1,13 @@
 {
-  "extends": "./node_modules/gts/tsconfig-google.json",
+  "extends": "./tsconfig.full.json",
```
Comment (ofrobots) [inline tsconfig.json]: is tsconfig.full.json different from tsconfig-google.json? If so, what's the rationale here?

File: tsconfig.full.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "noImplicitAny": false,
+    "noImplicitThis": false,
+    "noEmitOnError": false
+  },
+  "include": [
+    "*.ts",
+    "src/**/*.ts",
+    "test/**/*.ts"
+  ]
+}
```
Comment (ofrobots) [inline tsconfig.full.json]: new line

File: tsconfig.json [hunk 5]
```diff
@@ -1,18 +1,13 @@
...
+    "noImplicitThis": true,
+    "noEmitOnError": true
   },
   "include": [
     "*.ts",
-    "src/**/*.ts",
-    "test/**/*.ts"
+    "src/*.ts"
   ]
 }
```
Comment (ofrobots) [inline tsconfig.json]: add new line while you're here.

File: scripts/index.ts (hunk 1 above)
Comment (kjin) [reply scripts/index.ts]: This is a build script -- wanted to support npm commands that have a hyphen in its name.

File: src/trace-api.ts (hunk 2 above)
Comment (kjin) [reply src/trace-api.ts]: `pluginName_` is an undocumented field. I would not consider this semver-major.

File: tsconfig.json (hunk 3 above)
Comment (kjin) [reply tsconfig.json]: Edited abstract above. Hope it makes sense

File: tsconfig.full.json (hunk 4 above)
Comment (kjin) [reply tsconfig.full.json]: Done

File: tsconfig.json (hunk 5 above)
Comment (kjin) [reply tsconfig.json]: Done

**Pull request merged**